### PR TITLE
Query Layers

### PR DESCRIPTION
**Pull request by stdavis**
Looking for feedback on basic query layer component as well as working with the state machine context. Is that the right place to keep the selected query layers state?

### CODE REVIEW

File: src/components/search-wizard/QueryLayer.jsx [hunk 1]
```diff
@@ -1,9 +1,96 @@
...
+              label="more info"
+              size="xs"
+            />
+          </a>
+        }
+      >
+        <div className="max-w-md">
+          {config[fieldNames.queryLayers.layerDescription]}
+        </div>
+      </Tooltip>
```
Comment (steveoh) [inline src/components/search-wizard/QueryLayer.jsx:78]: **thought**: This interaction seemed odd to me. I clicked the ? icon to find more information and it opened a new link. I expected a help icon, when clicked, to show help text. This seemed to have an external link icon behavior. 

Then I came back and slow hovered over it to find out more information about the layer. This interaction won't be accessible in a touch environment.

I wonder if the hover text and ? icon should be on the layer name and the metadata external link icon is either in the tooltip or a separate icon?

File: index.html [hunk 2]
```diff
@@ -15,7 +15,7 @@
 
     <title>Utah DEQ Environmental Interactive Map</title>
   </head>
-  <body class="h-full w-full overflow-hidden">
+  <body class="h-full w-full overflow-hidden text-gray-700">
```
Comment (steveoh) [inline index.html:18]: **suggestion**: it's a best practice to put the text colors on the elements that use them. This makes it easier to add a dark mode. Just a suggestion, feel free to resolve without action.

File: src/components/search-wizard/QueryLayer.jsx [hunk 3]
```diff
@@ -1,9 +1,106 @@
...
+  },
+  {}
+);
+QueryLayer.propTypes = {
+  config: PropTypes.shape(configShape),
+  selected: PropTypes.bool.isRequired,
+  onSelectedChange: PropTypes.func.isRequired,
+  // filter: PropTypes.object,
+  // onFilterChange: PropTypes.func.isRequired,
+};
```
Comment (steveoh) [inline src/components/search-wizard/QueryLayer.jsx:106]: **thought**: would this be better served as a yup schema that you can validate? I think proptypes are only shown during development.

**Review by steveoh — COMMENTED** [review]
I'm happy with your state machine usage. It looks great. The UX on the query layer component is a little odd with the tooltip and help icon.

File: src/components/search-wizard/QueryLayer.jsx (hunk 1 above)
Comment (stdavis) [reply src/components/search-wizard/QueryLayer.jsx:78]: That's a good point. I moved the hover popup to the layer name and then gave the external link a better icon.

File: src/components/search-wizard/QueryLayer.jsx (hunk 3 above)
Comment (stdavis) [reply src/components/search-wizard/QueryLayer.jsx:106]: Great idea. I've implemented this with yup.

File: index.html (hunk 2 above)
Comment (stdavis) [reply index.html:18]: Interesting. Maybe this should be part of the utds plugin. I'm going to try to move forward for now.

**Pull request merged**